packages/sdk-cli/src/models/localRelay/relayInfo.ts

```diff
@@ -0,0 +1,67 @@
+import { cwd } from 'process';
+import { join } from 'path';
+import waitFor from 'p-wait-for';
+
+import { isInt, readJsonFile } from './util';
+import { RELAY_PKG_NAME, RELAY_PID_FILE_PATH } from './const';
+
+export type RelayInfo = { port: number; pid: number };
+
+export type ReadRelayInfoResult = RelayInfo | false;
+
+export async function readRelayInfo(): Promise<ReadRelayInfoResult> {
+  try {
+    // Error will be thrown and caught if readJsonFile returns undefined or anything else that can't be destructured
+    const { port, pid } = (await readJsonFile(
+      RELAY_PID_FILE_PATH,
+    )) as RelayInfo;
+
+    // [port, pid].every(...) doesn't pass Control Flow Analysis. I.e. TS won't know port & pid are numbers.
+    if (isInt(port) && isInt(pid)) {
+      return {
+        port: Number(port),
+        pid: Number(pid),
+      };
+    }
+  } catch (error) {
+    if ((error as NodeJS.ErrnoException).code !== 'ENOENT') {
+      console.error(error);
+    }
+  }
+
+  return false;
+}
+
+export async function pollRelayInfo(
+  timeout = 15000,
+  interval = 300,
+  readRelayInfoFn = readRelayInfo,
```

Comment:
Why does this need to be an argument? The extra indirection just makes it a little harder to follow.

Reply:
In `pollRelayInfo` tests I sometimes need to mock `readRelayInfo`. And I don't really want to separate `readRelayInfo.ts` into 2 separate files

Reply:
Just mock the file it's reading underneath?

Reply:
I test for the number of times `readRelayInfoFn()` has been called, too

Reply:
https://github.com/Fitbit/developer-bridge/pull/176/commits/27f8d0133fbe6519f135b8f278e0f089a5aa60b3